app/views/courses/HeroSelectModal.coffee

```diff
@@ -0,0 +1,42 @@
+ModalView = require 'views/core/ModalView'
+template = require 'templates/courses/hero-select-modal'
+Classroom = require 'models/Classroom'
+CocoCollection = require 'collections/CocoCollection'
+State = require 'models/State'
+ThangType = require 'models/ThangType'
+User = require 'models/User'
+
+module.exports = class HeroSelectModal extends ModalView
+  id: 'hero-select-modal'
+  template: template
+
+  events:
+    'click .select-hero-btn': 'onClickSelectHeroButton'
+    'click .hero-option': 'onClickHeroOption'
+
+  initialize: ({ currentHeroID }) ->
+    @state = new State({
+      currentHeroID
+      selectedHeroID: currentHeroID
+    })
+
+    @heroes = new CocoCollection([], {model: ThangType})
+    @heroes.url = '/db/thang.type?view=heroes'
+    @heroes.setProjection ['original', 'name', 'heroClass']
+    @heroes.comparator = 'gems' # TODO: Random? Alphabetical? Something else?
+    @supermodel.loadCollection(@heroes, 'heroes')
+
+    @listenTo @state, 'all', @render
+    @listenTo @heroes, 'all', @render
+
+  onClickHeroOption: (e) ->
+    heroID = $(e.currentTarget).data('hero-id')
+    @state.set selectedHeroID: heroID
+    hero = @heroes.get(heroID)
+    me.set(heroConfig: {}) unless me.get('heroConfig')
+    me.get('heroConfig').thangType = hero.get('original')
+    me.save().then =>
+      @trigger 'hero-select:success', hero
```

Comment:
Where is that doc? Not sure I've seen it.

This behavior was a decision I made when I was including all the heroes and the window was rather large — the button on the bottom wasn't even visible on smaller screens. Figured people wouldn't necessarily notice the button.

I'm not firmly set on this behavior, but having to explicitly save it feels like unnecessary mental overhead. But It does make it inconsistent with other parts of our website, so that's not great.